src/modules/Packages.rb

```diff
@@ -1047,26 +1048,57 @@ def boardPackages
       packages
     end
 
-    # Compute packages required to access the repository
+    # Compute packages required to access the repositories
     # @return [Array](string) list of the required packages
     def sourceAccessPackages
-      # TODO: rather check all registered repositories...
-      ret = []
+      packages = []
+      schemes = repo_schemes
 
-      instmode = Linuxrc.InstallInf("InstMode")
-      Builtins.y2milestone("Installation mode: %1", instmode)
+      # /sbin/mount.cifs is required to mount a SMB/CIFS share
+      packages << "cifs-mount" if schemes.include?("smb") || schemes.include?("cifs")
 
-      if instmode == "smb" || instmode == "cifs"
-        # /sbin/mount.cifs is required to mount a SMB/CIFS share
-        ret = ["cifs-mount"]
-      elsif instmode == "nfs"
-        # portmap is required to mount an NFS export
-        ret = ["nfs-client"]
-      end
+      # portmap is required to mount an NFS export
+      packages << "nfs-client" if schemes.include?("nfs")
 
-      Builtins.y2milestone("Packages for accessing the repository: %1", ret)
+      log.info("Packages for accessing the repositories: #{packages.inspect}")
+      packages
+    end
 
-      deep_copy(ret)
+    # Return the URL schemes for the currently defined repositories (only enabled
+    # repositories are evaluated). For ISO repositories the base URL schema
+    # is returned (i.e. the location of the ISO), invalid URLs are ignored.
+    # @return [Array<String>] the list of the URL schemes, empty if
+    #   no repository is defined
+    def repo_schemes
+      schemes = []
+
+      # all enabled repositories
+      Pkg.SourceGetCurrent(true).map do |repo|
```

Comment:
As you are not doing anything with the return value, you could use a plain `each` or, if you prefer, a `each_with_object`.

Reply:
Oh, originally I started the code with `map`, but it turned out that there is no mapping for some (invalid) values and I'd like to avoid duplicates. So in the end I used a different solution.